workers/unity/Assets/Gdk/Core/Systems/SpatialOSSendSystem.cs

```diff
@@ -13,11 +16,18 @@ internal static class Warnings
                 "Custom Replication System for type {0} already exists!";
         }
 
+        internal struct ComponentReplicator
+        {
+            public ComponentReplication ComponentReplication;
+            public ComponentGroup BasicReplicationGroup;
+            public List<ComponentGroup> CommandReplicationGroups;
+        }
+
         private WorkerBase worker;
         private MutableView view;
 
-        private readonly List<int> registeredReplicators = new List<int>();
-        private readonly List<int> commandSenders = new List<int>();
+
+        private readonly Dictionary<uint, ComponentReplicator> componentReplicators = new Dictionary<uint, ComponentReplicator>();
```

Comment:
Potential minor optimisation - make this a list. Removal is something we only do at the start and will probably only be done a couple of times. Normally we are just iterating over the whole thing.

Reply:
Yup, makes sense.